feature-#164047025-User should be able to login with Fa…

kelvinndmo opened the pull request:
## Description ##
This setups up social authentication with Google and Facebook to the application

## Type of change ##
Please select the relevant option
- [ ] Bug fix(a non-breaking change which fixes an issue)
- [x] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested? ##
Please describe the tests that you ran to verify your changes
- [ ] End to end test
- [ ] Integration test
- [x] unit test

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## Related Pivotal Tracker stories ##
[#164047025](https://www.pivotaltracker.com/story/show/164047025)

## Reviewers ##
- @andela/olympian-devs
- @Derrickkip 

#Screenshots

<img width="1440" alt="Screen Shot 2019-04-16 at 4 10 41 PM" src="https://user-images.githubusercontent.com/32582512/56213337-7ce6b000-6064-11e9-9556-b60550511f84.png">

Commit:
164047025-feature-(social auth): User should be able to login with Facebook,Google and Twitter.

-User should be able to login with Facebook,Google and Twitter
-Test for the feature

[Finishes #164047025]

mnswaleh reviewed (commented):
place the social login buttons on the login page

kelvinndmo commented:
> place the social login buttons on the login page

@mnswaleh done.Thanks for the review

Kibetchirchir commented on package.json:
```diff
@@ -35,7 +46,7 @@
     "start": "npm-run-all -p watch-css start-js",
     "build": "npm run build-css && react-scripts build",
-    "start-js": "react-scripts start",
+    "start-js": "PORT=4000 react-scripts start",
```
remove the port number please

kevpy commented on src/components/Home/__snapshots__/App.test.js.snap:
```diff
@@ -0,0 +1,17 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
This file should not be tracked. Please remove it.

kevpy commented on src/components/Home/Home.css:
```diff
@@ -0,0 +1,10 @@
+.App-header {
```
Please avoid tracking `.css` files. We are using `Sass`. In this context please remove it form files being tracked by git

kevpy commented on src/components/social/styles.css:
```diff
@@ -0,0 +1,53 @@
+.novak {
```
Also, add this file to the file that need not be tracked

kelvinndmo replied:
noted working on it

kevpy commented on src/components/Login/Login.test.js:
```diff
@@ -1,98 +1,92 @@
-import loginAction from '../../actions/index';
-import { Login } from './Login';
-import store from '../../store/store';
+import React from 'react'
```
Please undo the formatting of the code added. Absence of trailing commas and semi-colons will cause hound to flag the current PR and any other PR that contains the same file as not meeting `eslint` rules. This applies to the `Login.js` file as well

kelvinndmo replied:
noted

kevpy commented on src/components/Login/Login.css:
```diff
@@ -0,0 +1,99 @@
+body {
```
Please remove this file. It shouldn't be tracked.

kevpy commented on src/actions/auth.js:
```diff
@@ -0,0 +1,103 @@
+// import axios from 'axios';
```
If this file isn't needed in your PR, please remove it. It is commented out, thus this means it is not needed.

Derrickkip commented on src/components/Login/Login.js:
```diff
@@ -135,9 +136,9 @@ export class Login extends Component {
-                {/*)}*/}
+                {/* {formErrors.password.length > 0 && ( */}
+                {/*  <span className="errorMessage">{formErrors.password}</span> */}
+                {/* )} */}
```
Remove the commented lines

Derrickkip commented on src/components/social/index.js:
```diff
@@ -0,0 +1,104 @@
+    return (
+      <SocialButton
+        className={`waves-effect waves-light btn-flat btn--default m-b--15 button social-button ${provider}`}
+        id="#facebook-connect"
```
Having this id Here causes there to be two buttons with the same ID, you should pass it as props

Derrickkip commented on src/components/social/index.js:
```diff
@@ -0,0 +1,104 @@
+};
+
+SocialAuth.propTypes = {
+  history: PropTypes.shape({
```
Why do you have two descriptions of the propTypes?

Commit:
164047025-feature-(social auth): User should be able to login with Facebook,Google and Twitter.

-User should be able to login with Facebook,Google and Twitter
-Test for the feature

[Finishes #164047025]
